dali/test/python/test_operator_normalize.py

```diff
@@ -0,0 +1,391 @@
+            ref_scalar_mean_ext = normalize_list(batch, data, axes, mean = 1, stddev = stddev)
+            ref_scalar_stddev_ext = normalize_list(batch, data, axes, mean = mean, stddev = 2)
+
+            shift_scale(ref_ext_mean, shift, scale)
+            shift_scale(ref_ext_stddev, shift, scale)
+            shift_scale(ref_ext_all, shift, scale)
+            shift_scale(ref_scalar_mean_ext, shift, scale)
+            shift_scale(ref_scalar_stddev_ext, shift, scale)
+
+            check(ext_mean, ref_ext_mean)
+            check(ext_stddev, ref_ext_stddev)
+            check(ext_all, ref_ext_all)
+            check(scalar_mean_ext, ref_scalar_mean_ext)
+            check(scalar_stddev_ext, ref_scalar_stddev_ext)
+
+        if scalar_params is not None:
+            ref_scalar_params = normalize_list(batch, data, axes, mean = 1, stddev = 2)
+            shift_scale(ref_scalar_params, shift, scale)
+            check(scalar_params, ref_scalar_params)
+
+    def iter_setup(self):
+        self.feed_input(self.input_data, generate_data(self.dims, self.batch_size, self.batch, self.axes, dtype = self.in_type))
+
+def to_list(tensor_list):
+    out = []
+    for i in range(len(tensor_list)):
+        out.append(tensor_list.at(i))
+    return out
+
+np.random.seed(seed=1337)
+
+def mask2axes(mask):
+    out = []
+    a = 0
+    while mask:
+        if mask & 1:
+            out.append(a)
+        mask >>= 1
+        a += 1
+    return out
+
+def all_axes(dim):
+    yield None
+    for mask in range(1, 1 << dim):
+        yield mask2axes(mask)
+
+def _run_test(device, batch_size, dim, axes, axis_names, batch_norm,
+             out_type = None, in_type = None, shift = None, scale = None):
+    kind = "inter-sample" if batch_norm else "per-sample"
+    msg = "{0}, {1}, batch = {2}, dim = {3}".format(device, kind, batch_size, dim)
+    if axes is not None:
+        msg += " axes = {}".format(axes)
+    if axis_names is not None:
+        msg += " axis_names = '{}'".format(axis_names)
+    if out_type is not None:
+        msg += " output = {}".format(out_type)
+    if in_type is not None:
+        msg += " input = {}".format(in_type)
+    print(msg)
+
+    pipe = NormalizePipeline(device, batch_size, dim, axes, axis_names, batch_norm, out_type, in_type, shift, scale)
+    pipe.build()
+    out = pipe.run()
+    pipe.check_batch(*[to_list(x) for x in out])
+
+def axes2names(axes, layout='abcdefghijklmnopqrstuvwxyz'):
+    return "".join([layout[axis] for axis in axes])
+
+def _test_up_to_5D_all_axis_combinations(device):
+    batch_size = 10
+    for batch_norm in [False, True]:
+        for dim in range(1, 6):
+            for axes in all_axes(dim):
+                yield _run_test, device, batch_size, dim, axes, None, batch_norm
+                if axes is not None and dim < 5:
+                    axis_names = axes2names(axes)
+                    yield _run_test, device, batch_size, dim, None, axis_names, batch_norm
```

Comment:
I'm afraid that yield from nested function doesn't work with nose. You need to have it inside the main test case - test_cpu_up_to_5D_all_axis_combinations (or at least it didn't work for me - silent success without running anything).

Reply:
But this is somehow different and actually might work.

Reply:
This does work, but you need to iterate and yield in the outer function (see below).